File: ruby/mongo_sharded_secure.rb
# Connecting to a sharded instance using SSL:
# Make sure to change the port number when using an SSL connection.

#!/usr/bin/env ruby
require 'mongo'

# Turn off debug-mode
Mongo::Logger.logger.level = Logger::WARN

client_host = ['iad1-mongos0.objectrocket.com:266073']
client_options = {
  database: 'tedst_db',
  user: 'sooz',
  password: 'xxxxxxxx',
  ssl: true
}

client = Mongo::Client.new(client_host, client_options)

puts('Client Connection: ')
puts(client.cluster.inspect)
puts
puts('Collection Names: ')
puts(client.database.collection_names)

File: python/mongo_sharded_secure.py
# Example Documenetation: https://objectrocket.com/docs/mongodb_python_examples.html#connecting (modified version)
# Pymonogo Documentation: https://api.mongodb.com/python/current/
# Assumed you have run 'pip install pymongo'
import pymongo

settings = {
    'host': 'iad1-mongos0.objectrocket.com:26139',
    'database': 'test_db',
    'username': 'sooz',
    'password': 'xxxxxxxx',
    'options': 'ssl=true'
}

connection_string = "mongodb://{username}:{password}@{host}/{database}?{options}".format(**settings);

try:
    conn = pymongo.MongoClient(connection_string)
    collectionNames = conn[settings['database']].collection_names() # This line actually triggers something that test the credentials
    print "Connected; collection Names", collectionNames
    conn.close(); # Actually close the connection
except Exception as ex:
    print "Error:", ex
    exit('Failed to connect, terminating.')
